rustup-init.sh

```diff
@@ -8,14 +9,20 @@
 # It runs on Unix shells like {a,ba,da,k,z}sh. It uses the common `local`
 # extension. Note: Most shells limit `local` to 1 var per line, contra bash.
 
-if [ "$KSH_VERSION" = 'Version JM 93t+ 2010-03-05' ]; then
-    # The version of ksh93 that ships with many illumos systems does not
-    # support the "local" extension.  Print a message rather than fail in
-    # subtle ways later on:
-    echo 'rustup does not work with this ksh93 version; please try bash!' >&2
-    exit 1
-fi
+# Some versions of ksh have no `local` keyword. Alias it to `typeset`, but
+# beware this makes variables global with f()-style function syntax in ksh93.
+# mksh has this alias by default.
+has_local() {
+    # shellcheck disable=SC2034  # deliberately unused
+    local _has_local
+}
 
+has_local 2>/dev/null || alias local=typeset
+
+# zsh does not split words by default, Required for curl retry arguments below.
+if [ -n "$ZSH_VERSION" ]; then
+    setopt shwordsplit
```

Comment:
Concern: I don't use `zsh` personally, but does this make persistent changes somehow? (Possibly related: https://github.com/sdkman/sdkman-cli/issues/210)

Reply:
It seems to me that one solution might be calling `setopt local_options shwordsplit` (I'm looking at [`nvm.sh`](https://github.com/nvm-sh/nvm/blob/8a83b36688e270e460803ab842e3333393b523c7/nvm.sh#L1497), for example).

Reply:
Good point. If the script is directly `source`d, then yes it would persist for as long as that shell session is alive. I'm not sure if that's something that's recommended though, rather then executing via `zsh rustup-init.sh` or as on [rustup.rs](https://rustup.rs/): `curl ... rustup-init.sh | zsh`.

Reply:
> It seems to me that one solution might be calling `setopt local_options shwordsplit` in a function (I'm looking at [`nvm.sh`](https://github.com/nvm-sh/nvm/blob/8a83b36688e270e460803ab842e3333393b523c7/nvm.sh#L1497), for example).
Yeah, that sounds better. I'll take a look.

Reply:
I see. Thanks for the clarification!

Reply:
@alexhudspith Oops I have a bad network connection here, sorry for the delay 😅